### DISCUSSION TITLE
How to infer the return type of query based on the params passed - RTK Query ?

### DISCUSSION THREAD

**Discussion by LoyalPotato** [reduxjs/redux-toolkit#2765] [Q&A] [answered]
Hello,

This was previously asked here #2701 but with no answers.

I just wanted to know if there has been any development that allows this or if it's not possible at all.

**Comment by phryneas** [comment, marked as answer]
It's not possible. That's not really an RTKQ limitation, it's a TypeScript limitation: we need to map over those types to create new functions, and at that point is is just not possible to have any connection between `ArgType` and `ReturnType` .

**Reply by LoyalPotato** [reply]
I see. Thanks for the quick answer 👍
